### ISSUE TITLE
Infinite loop on "Check link" with YouTube

### ISSUE THREAD

**Issue by NathanBnm** [torikulhabib/mindi#37]
When I tried to download a YouTube video I have a infinite loading on "Check link"

**Comment by torikulhabib** [comment]
https://github.com/torikulhabib/mindi/commit/a5707271e42c991344b1fc63b0f76022b11929ca

**Comment by NathanBnm** [comment]
@torikulhabib this commit didn't solve the issue on my side

**Comment by torikulhabib** [comment]
https://github.com/torikulhabib/mindi/commit/9b1c13272adbd6ef8c4573e86ee50bfa0f9a3b2f

**Comment by NathanBnm** [comment]
Now the spinner stops after a while, but nothing happens, I have no other choice to cancel.
The app doesn't get any info about the video.

**Comment by ryonakano** [comment]
![Screenshot from 2019-08-18 19-16-46](https://user-images.githubusercontent.com/26003928/63223190-c995ef80-c1ec-11e9-888d-af3bdc5e3c1b.png)

I installed the latest version of youtube-dl in the way described in [their readme](https://github.com/ytdl-org/youtube-dl/blob/master/README.md#installation) and build/installed the `master` of this repository, and I can also confirm this issue.

According to the output in the terminal:

```
** (com.github.torikulhabib.mindi:30740): WARNING **: 19:16:42.207: CheckLink.vala:84: Failed to change to directory “/home/ryo/.cache/com.github.torikulhabib.mindi” (No such file or directory)
```

The app does not create the cache directory automatically if it does not exist. I thought I've fixed something similar bug in #23, but that fix does not seem to be effective.
